seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java

```diff
@@ -369,30 +477,59 @@ private void restoreJobFromMasterActiveSwitch(@NonNull Long jobId, @NonNull JobI
         }
 
         String jobFullName = jobMaster.getPhysicalPlan().getJobFullName();
-        runningJobMasterMap.put(jobId, jobMaster);
-
+        if (isJobPending) {
+            pendingJobMasterMap.put(jobId, new Tuple2<>(PendingSourceState.RESTORE, jobMaster));
+        } else {
+            runningJobMasterMap.put(jobId, jobMaster);
+        }
         logger.info(
                 String.format(
                         "The restore %s is in %s state, restore pipeline and take over this job running",
                         jobFullName, jobStatus));
-        CompletableFuture.runAsync(
-                () -> {
-                    try {
-                        jobMaster
-                                .getPhysicalPlan()
-                                .getPipelineList()
-                                .forEach(SubPlan::restorePipelineState);
-                        jobMaster.run();
-                    } finally {
-                        // voidCompletableFuture will be cancelled when zeta master node
-                        // shutdown to simulate master failure,
-                        // don't update runningJobMasterMap is this case.
-                        if (!jobMaster.getJobMasterCompleteFuture().isCompletedExceptionally()) {
-                            runningJobMasterMap.remove(jobId);
+        // FIFO strategy，If there is a waiting task, the subsequent task will keep waiting
+        boolean canRunJob, preApplyResources;
+        if (isJobPending) {
+            preApplyResources = jobMaster.preApplyResources();
+            // The pending queue is empty and there are sufficient resources to run directly
+            canRunJob = (pendingJob.size() == 0 && preApplyResources);
+        } else {
+            // If it is a dynamic slot or the pending queue has not been started, run it directly
+            canRunJob = true;
+            preApplyResources = true;
+        }
+        if (canRunJob) {
+            CompletableFuture.runAsync(
+                    () -> {
+                        try {
+                            if (isJobPending) {
+                                pendingJobMasterMap.remove(jobId);
+                                runningJobMasterMap.put(jobId, jobMaster);
+                            }
+                            jobMaster
+                                    .getPhysicalPlan()
+                                    .getPipelineList()
+                                    .forEach(SubPlan::restorePipelineState);
+                            if (!dynamicSlot && !preApplyResources) {
```

Comment:
I can not find when the `dynamicSlot` and `preApplyResources` both false when `canRunJob` is true.